### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by mberezhna**
[DEMO LINK](https://mberezhna.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,70 @@
...
+    setErrorMessage,
+  } = useContext(AppContext);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMessage(ErrorType.UnableToLoadTodo));
+  }, [setErrorMessage, setTodos]);
+
+  const celectFilterTodo = useCallback(
```
Comment (MrHunterBoi) [inline src/App.tsx:31]: Typo

File: src/AppContext.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        .filter(todoToFind => todoToFind.completed);
+
+      setSelectedTodoIds(currentIds => (
+        [...currentIds, ...completedTodos
+          .map(completedTodo => completedTodo.id),
+        ]
+      ));
+
+      completedTodos.map(completedTodo => todoService
+        .deletTodo(completedTodo.id));
```
Comment (MrHunterBoi) [inline src/AppContext.tsx:79]: Typo

File: src/AppContext.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+      userId: USER_ID,
+      title,
+      completed: false,
+    });
+
+    todoService.createTodo({
+      userId: USER_ID,
+      title,
+      completed: false,
+    })
```
Comment (MrHunterBoi) [inline src/AppContext.tsx:103]: You may as well create a variable that contains data about the todo without id and use it in both instances

File: src/AppContext.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    clearCompleted,
+    setSelectedTodoIds,
+    todoTitle,
+    isLoading,
+    createNewTodo,
+    setTodoTitle,
+    tempTodo,
+    deleteTodo,
+    selectedTodoIds,
+  ]);
```
Comment (MrHunterBoi) [inline src/AppContext.tsx:186]: 👀

File: src/components/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  } = useContext(AppContext);
+
+  const todoTitleRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (todoTitleRef.current && !todoTitleRef.current.disabled) {
+      todoTitleRef.current.focus();
+    }
+  });
```
Comment (MrHunterBoi) [inline src/components/Header.tsx:27]: Afaik it should also autofocus when you create or delete a todo

**Review by MrHunterBoi — APPROVED** [review]
Good one

File: src/App.tsx (hunk 1 above)
Comment (roman-mirzoian) [reply src/App.tsx:31]: Good point

File: src/AppContext.tsx [hunk 6]
```diff
@@ -0,0 +1,193 @@
+/* eslint-disable @typescript-eslint/no-redeclare */
+/* eslint-disable no-redeclare */
+/* eslint-disable react-hooks/exhaustive-deps */
```
Comment (roman-mirzoian) [inline src/AppContext.tsx:3]: Try to rewrite the code so that it is not necessary to disable the linter

File: src/AppContext.tsx (hunk 2 above)
Comment (roman-mirzoian) [reply src/AppContext.tsx:79]: Good point

File: src/AppContext.tsx [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+      todoService.deletTodo(todoId)
+        .then(() => {
+          setTimeout(() => {
+            setTodos(currentTodos => currentTodos
+              .filter(post => post.id !== todoId));
+          }, 500);
+        })
+        .catch(() => {
+          setSelectedTodoIds(ids => {
+            ids.splice(ids.indexOf(todoId), 1);
```
Comment (roman-mirzoian) [inline src/AppContext.tsx:138]: Use `filter` instead if `splice` and `indexOf`

File: src/AppContext.tsx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+            return ids;
+          });
+          setErrorMessage(ErrorType.UnableToDeleteTodo);
+          setTimeout(() => setErrorMessage(null), 2000);
+        })
+        .finally(() => setTimeout(
+          () => setSelectedTodoIds(ids => ids.splice(ids.indexOf(todoId), 1)),
```
Comment (roman-mirzoian) [inline src/AppContext.tsx:146]: `filter`

File: src/AppContext.tsx (hunk 4 above)
Comment (roman-mirzoian) [reply src/AppContext.tsx:186]: @MrHunterBoi better to leave more meaningful comments

**Review by roman-mirzoian — CHANGES_REQUESTED** [review]
Good job, but there is quite a bit of work to be done.

**Review by roman-mirzoian — APPROVED** [review]
Well done 👍

**Pull request closed without merging**